src/libraries/System.Private.CoreLib/src/System/IO/BinaryReader.cs

```diff
@@ -63,16 +64,14 @@ public BinaryReader(Stream input, Encoding encoding, bool leaveOpen)
             {
                 minBufferSize = 16;
             }
-
-            _buffer = new byte[minBufferSize];
-            // _charBuffer and _charBytes will be left null.
+            _bufferSize = minBufferSize;
```

Comment:
If I'm reading the code correctly, we'll now never use the _buffer outside of FillBuffer and never consume the data read into the buffer.  Can't we just delete _bufferSize and _buffer entirely, and just use ArrayPool in FillBuffer if/when we need a temporary array?

Reply:
Didn't think of that, thanks. `FillBuffer` has also been refactored to avoid it if length is zero or one.